packages/p2p/src/components/app.jsx

```diff
@@ -14,38 +14,41 @@ const App = props => {
     const { general_store, order_store } = useStores();
     const { className, history, lang, order_id, server_time, websocket_api } = props;
 
-    React.useEffect(() => {
-        general_store.setAppProps(props);
-        general_store.setWebsocketInit(websocket_api, general_store.client.local_currency_config.decimal_places);
-        order_store.setOrderId(order_id);
-
-        // Redirect back to /p2p, this was implemented for the mobile team. Do not remove.
-        if (/\/verification$/.test(history?.location.pathname)) {
-            localStorage.setItem('is_verifying_p2p', true);
-            history.push(routes.cashier_p2p);
-        }
+    const isMounted = useIsMounted();
 
-        setLanguage(lang);
-        ServerTime.init(server_time);
+    React.useEffect(() => {
+        if (isMounted()) {
+            general_store.setAppProps(props);
+            general_store.setWebsocketInit(websocket_api, general_store.client.local_currency_config.decimal_places);
+            order_store.setOrderId(order_id);
 
-        // force safari refresh on back/forward
-        window.onpageshow = function (event) {
-            if (event.persisted) {
-                window.location.reload();
+            // Redirect back to /p2p, this was implemented for the mobile team. Do not remove.
+            if (/\/verification$/.test(history?.location.pathname)) {
+                localStorage.setItem('is_verifying_p2p', true);
+                history.push(routes.cashier_p2p);
             }
-        };
 
-        waitWS('authorize').then(() => {
-            general_store.onMount();
-            if (localStorage.getItem('is_verifying_p2p')) {
-                localStorage.removeItem('is_verifying_p2p');
-                general_store.setActiveIndex(general_store.path.my_ads);
-            }
-        });
+            setLanguage(lang);
+            ServerTime.init(server_time);
+
+            // force safari refresh on back/forward
+            window.onpageshow = function (event) {
+                if (event.persisted) {
+                    window.location.reload();
+                }
+            };
 
+            waitWS('authorize').then(() => {
+                general_store.onMount();
+                if (localStorage.getItem('is_verifying_p2p')) {
+                    localStorage.removeItem('is_verifying_p2p');
+                    general_store.setActiveIndex(general_store.path.my_ads);
+                }
+            });
+        }
         return () => general_store.onUnmount();
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    }, [isMounted]);
```

Comment:
```suggestion
    }, []);
```
we should remove `isMounted` from the dependency list because `isMounted` is an anonymous function created on each render.

```
export const useIsMounted = () => {
    const is_mounted = React.useRef(false);

    React.useEffect(() => {
        is_mounted.current = true;

        return () => {
            is_mounted.current = false;
        };
    }, []);
    return () => is_mounted.current;
};
```

Reply:
Let me check this again